tests/binning_test.py

```diff
@@ -944,3 +944,52 @@ def test_group_many_groups(ngroup):
     del expected.coords['label']
     assert_identical(grouped.bins.constituents['data'][::2], expected)
     assert_identical(grouped.bins.constituents['data'][1::2], expected)
+
+
+@pytest.fixture
+def noncontiguous_var() -> sc.Variable:
+    var = sc.linspace('x', 0, 1, 101, unit='m')[::2]
+    assert not var.values.data.contiguous
+    return var
+
+
+@pytest.fixture
+def contiguous_var(noncontiguous_var: sc.Variable) -> sc.Variable:
+    var = noncontiguous_var.copy()
+    assert var.values.data.contiguous
+    return var
+
+
+def test_noncontiguous_binning(
+    noncontiguous_var: sc.Variable, contiguous_var: sc.Variable
+):
+    table = sc.data.table_xyz(1_000)
+    assert sc.identical(table.bin(x=noncontiguous_var), table.bin(x=contiguous_var))
```

Comment:
This test fails, but I don't know why it fails...

Here is the code snippet to reproduce this:
```python
table = sc.data.table_xyz(10_000)
noncont_var = sc.linspace('x', 0, 1, 101, unit='m')[::2]
con_var = noncont_var.copy()
assert sc.identical(noncont_var, con_var)
assert not sc.identical(table.bin(x=noncont_var), table.bin(x=con_var))
```
And the first two bins of each binned data array were different like this:
![image](https://github.com/scipp/scipp/assets/17974113/bf56282c-8a06-4a53-9bce-45afd98cdc1d)


```

Reply:
This one is now fixed...!